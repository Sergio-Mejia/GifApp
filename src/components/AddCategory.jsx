import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    //No aceptar una letra o enviar mensajes sin contenido
    if (inputValue.trim().length <= 1) return;

    //Obtener las categorias anteriores y agregar una nueva
    // setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());

    //Limpiar caja de texto
    setInputValue("");
  };

  return (
    <form action="" onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifts"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
