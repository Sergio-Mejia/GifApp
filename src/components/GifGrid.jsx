import { useState, useEffect } from "react";

import { GifItem } from "./GifItem";
import { getGifts } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  const getImages = async () => {
    const newImage = await getGifts(category);
    setImages(newImage);
  };

  useEffect(() => {
    getImages();
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((image) => (
          <GifItem 
            key={ image.id }
            { ...image } 
          />
        ))}
      </div>
    </>
  );
};
