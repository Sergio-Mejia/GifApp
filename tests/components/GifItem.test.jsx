import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem", () => {
  const title = "React";
  const url = "https://react.com/";

  test("DeberÃ­a hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should mostrar la imagen con la Url y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // console.log(screen.getByRole('img').src);
    // expect( screen.getByRole('img').src ).toBe( url );
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("should mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect( screen.getByText( title ) ).toBeTruthy();
  });
});
