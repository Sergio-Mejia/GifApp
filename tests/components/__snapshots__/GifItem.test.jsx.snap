// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Deber√≠a hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="React"
      src="https://react.com/"
    />
    <p>
      React
    </p>
  </div>
</div>
`;
